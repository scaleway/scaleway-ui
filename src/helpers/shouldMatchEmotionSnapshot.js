/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
