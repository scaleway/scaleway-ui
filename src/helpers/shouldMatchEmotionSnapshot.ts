/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
