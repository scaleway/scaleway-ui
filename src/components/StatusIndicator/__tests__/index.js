import React from 'react'
import { StatusIndicator, statuses } from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'

describe('StatusIndicator', () => {
  statuses.forEach(status => {
    test(`render ${status}`, () => {
      shouldMatchEmotionSnapshot(<StatusIndicator status={status} />)
    })
  })

  test(`render unknow`, () => {
    const spy = jest.spyOn(console, 'error').mockImplementation()

    shouldMatchEmotionSnapshot(<StatusIndicator status="unknow" />)
    expect(console.error).toHaveBeenCalledTimes(1)
    spy.mockRestore()
  })
})
