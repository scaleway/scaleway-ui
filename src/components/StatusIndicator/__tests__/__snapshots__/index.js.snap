// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIndicator render animated 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.1;
  }

  100% {
    opacity: 1;
  }
}

.emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #45d6b5;
  -webkit-animation: animation-0 linear 1s infinite;
  animation: animation-0 linear 1s infinite;
}

<div
    class="emotion-0 emotion-1"
    status="available"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render available 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #45d6b5;
}

<div
    class="emotion-0 emotion-1"
    status="available"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render creating 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="creating"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render deleting 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="deleting"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render deployed 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #45d6b5;
}

<div
    class="emotion-0 emotion-1"
    status="deployed"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render disk_full 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #ef5774;
}

<div
    class="emotion-0 emotion-1"
    status="disk_full"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render error 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #ef5774;
}

<div
    class="emotion-0 emotion-1"
    status="error"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render locked 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #ef5774;
}

<div
    class="emotion-0 emotion-1"
    status="locked"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render pending 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="pending"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render ready 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #45d6b5;
}

<div
    class="emotion-0 emotion-1"
    status="ready"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render running 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #45d6b5;
}

<div
    class="emotion-0 emotion-1"
    status="running"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render snapshotting 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="snapshotting"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render starting 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="starting"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render stopped 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #b2b6c3;
}

<div
    class="emotion-0 emotion-1"
    status="stopped"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render stopped_in_place 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #ff8c69;
}

<div
    class="emotion-0 emotion-1"
    status="stopped_in_place"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render stopping 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="stopping"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render unavailable 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #ef5774;
}

<div
    class="emotion-0 emotion-1"
    status="unavailable"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render unknow 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="unknow"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render updating 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="updating"
  />
</DocumentFragment>
`;

exports[`StatusIndicator render warning 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  background-color: #3f6ed8;
}

<div
    class="emotion-0 emotion-1"
    status="warning"
  />
</DocumentFragment>
`;
